### PR TITLE
fix(FEC-11136): Tizen 4 got stuck on the beginning of the media

### PR DESCRIPTION
**Pull request by Yuvalke**
### Description of the Changes

Issue: Tizen 4 has stalled on the beginning of the media/after first seek we want to make sure the media wouldn't stop.
Solution: make a small seeks until playback keeps playing.

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/dash-adapter.js [hunk 1]
```diff
@@ -365,10 +372,42 @@ export default class DashAdapter extends BaseMediaSourceAdapter {
...
+      this._stallInterval = null;
+    }
+  }
+
+  _stallSmartTVHandler(): void {
+    this._clearStallInterval();
+    let lastCurrentTime = this._videoElement.currentTime;
+    this._stallInterval = setInterval(() => {
+      if (lastCurrentTime === this._videoElement.currentTime) {
+        this._videoElement.currentTime = parseFloat(this._videoElement.currentTime.toFixed(1)) + 0.1;
```
Comment (OrenMe) [inline src/dash-adapter.js:395]: move 0.1 to const, maybe expose as config?

File: src/dash-adapter.js [hunk 2]
```diff
@@ -365,10 +372,42 @@ export default class DashAdapter extends BaseMediaSourceAdapter {
...
+      } else {
+        this._clearStallInterval();
+      }
+    }, 2 * 1000);
```
Comment (OrenMe) [inline src/dash-adapter.js:399]: move to const, maybe expose as config?

File: src/dash-adapter.js [hunk 3]
```diff
@@ -365,10 +375,42 @@ export default class DashAdapter extends BaseMediaSourceAdapter {
...
+
+  /**
+   * register to event to break the stalls on smart TV
+   * @returns {void}
+   * @private
+   */
+  _maybeBreakStalls(): void {
+    if (this._config.playback.forceBreakStall) {
+      this._eventManager.listenOnce(EventType.STALLED, this._stallSmartTVHandler);
+      this._eventManager.listenOnce(EventType.SEEKED, this._stallSmartTVHandler);
```
Comment (OrenMe) [inline src/dash-adapter.js:410]: do we need/want to remove them after one handler is called?
If stalled occurred we can remove seeked and vide versa?

File: src/dash-adapter.js (hunk 2 above)
Comment (Yuvalke) [reply src/dash-adapter.js:399]: Didn't want to do it too robust cause I believe we need to handle it on Shaka, WDYT?

File: src/dash-adapter.js (hunk 3 above)
Comment (Yuvalke) [reply src/dash-adapter.js:410]: I clear the stall interval if it's not stopped for each one of them.
Yes, the issue that we saw occur only on the beginning.

File: src/dash-adapter.js (hunk 2 above)
Comment (Yuvalke) [reply src/dash-adapter.js:399]: To create 3 configs for hack we're adding for edge case that we won't need anymore after Shaka will handle it, it sounds too robust for me, don't you think?

Comment (OrenMe) [reply src/dash-adapter.js:399]: agree, let's start lean, but please move to consts and we will hope Shaka will approve this :-)

File: src/dash-adapter.js [hunk 4]
```diff
@@ -144,6 +158,20 @@ export default class DashAdapter extends BaseMediaSourceAdapter {
...
 
+  /**
+   * stall interval to break the stall on Smart TV
+   * @type {null|IntervalID}
+   * @private
+   */
+  _stallInterval: ?IntervalID = null;
+
+  /**
+   * stall interval to break the stall on Smart TV
```
Comment (yairans) [inline src/dash-adapter.js:169]: fix desc

**Review by yairans — CHANGES_REQUESTED** [review]

File: src/dash-adapter.js [hunk 5]
```diff
@@ -365,10 +396,43 @@ export default class DashAdapter extends BaseMediaSourceAdapter {
...
+  }
+
+  /**
+   * register to event to break the stalls on smart TV
+   * @returns {void}
+   * @private
+   */
+  _maybeBreakStalls(): void {
+    if (this._config.forceBreakStall) {
+      this._eventManager.listenOnce(this._videoElement, EventType.STALLED, () => this._stallSmartTVHandler());
```
Comment (yairans) [inline src/dash-adapter.js:432]: so you want to keep it once only?

**Review by yairans — APPROVED** [review]

**Pull request merged**